### PR TITLE
refactor: cleanup, add docs and merge ETL/hermite with ETL/bezier

### PR DESCRIPTION
**Pull request by rodolforg**

### CODE REVIEW

**Comment by rodolforg** [comment]
I just found out that there is another Bézier and Hermite classes in synfig-core o.O
I wonder if we can merge them...
https://github.com/synfig/synfig/blob/9cc796e08727bf478c87dc93a74e46822398a16b/synfig-core/src/synfig/curve.h

File: ETL/ETL/_bezier.h [hunk 1]
```diff
@@ -591,6 +506,31 @@ class bezier : public bezier_base<V,T>
...
+		return a < b ? a : b;
+	}
+
+	/** Find maximum of a and b */
+	template<typename U>
+	static U max(const U& a, const U& b)
+	{
+		return a > b ? a : b;
+	}
+
```
Comment (rodolforg) [inline ETL/ETL/_bezier.h:533]: @ice0 should I simply `#include <algorithm>` to use `std::min()` and `std::max()` or is it better like this?

Comment (ice0) [reply ETL/ETL/_bezier.h:533]: <algorithm> is about 400KB so I prefer the current solution

**Comment by rodolforg** [comment]
To merge or not to merge?
To what filename? XD

```
$ git grep \\bcurve\.h\\b
synfig-core/src/modules/mod_geometry/advanced_outline.cpp:#include <synfig/curve.h>
synfig-core/src/synfig/Makefile.am:     curve.h \
synfig-core/src/synfig/curve.cpp:#include "curve.h"
synfig-core/src/synfig/curve.h:/*!      \file curve.h
synfig-core/src/synfig/rendering/primitive/bend.cpp:#include <synfig/curve.h>
```
```
$ git grep \<ETL/bezier\\b
synfig-core/src/synfig/curve_helper.h:#include <ETL/bezier>
synfig-core/src/synfig/curveset.cpp:#include <ETL/bezier>
synfig-core/src/synfig/vector.h:#include <ETL/bezier>
```
```
$ git grep \<ETL/hermite\\b
ETL/test/hermite.cpp:#include <ETL/hermite>
synfig-core/src/modules/lyr_std/curvewarp.cpp:#include <ETL/hermite>
synfig-core/src/modules/mod_gradient/curvegradient.cpp:#include <ETL/hermite>
synfig-core/src/modules/mod_particle/plant.cpp:#include <ETL/hermite>
synfig-core/src/synfig/valuenodes/valuenode_animatedinterface.cpp:#include <ETL/hermite>
synfig-core/src/synfig/valuenodes/valuenode_bline.cpp:#include <ETL/hermite>
synfig-core/src/synfig/valuenodes/valuenode_blinecalctangent.cpp:#include <ETL/hermite>
synfig-core/src/synfig/valuenodes/valuenode_blinecalcvertex.cpp:#include <ETL/hermite>
synfig-core/src/synfig/valuenodes/valuenode_segcalctangent.cpp:#include <ETL/hermite>
synfig-core/src/synfig/valuenodes/valuenode_segcalcvertex.cpp:#include <ETL/hermite>
synfig-core/test/angle.cpp:#include <ETL/hermite>
synfig-core/test/benchmark.cpp:#include <ETL/hermite>
synfig-studio/src/gui/states/state_bline.cpp:#include <ETL/hermite>
synfig-studio/src/synfigapp/blineconvert.cpp:#include <ETL/hermite>
```

**Comment by rodolforg** [comment]
@ice0  I think this PR is now ready to merge.
Or should I move it to synfig-core before?

Please take special care on reviewing the penultimate commit (8245239b73fe1dbe53e1ce4b09a04416dd7a226e).

**Comment by ice0** [comment]
> To merge or not to merge?
> To what filename? XD

I suggest to leave it as-is for now. 

> Or should I move it to synfig-core before?

I think better to move it to `synfig-core` in the next iteration.

Looks good, please rebase.

**Comment by rodolforg** [comment]
Rebased

**Review by ice0 — APPROVED** [review]

**Comment by ice0** [comment]
Merged. Thank you!

**Pull request merged**